lib/__tests__/_sendEvent.test.ts

```diff
@@ -84,20 +94,189 @@ describe("sendEvent", () => {
     });
     it("should send the correct payload", () => {
       (AlgoliaInsights as any).sendEvent("click", {
-        objectID: "1"
+        eventName: "my-event",
+        indexName: "my-index",
+        objectID: "1",
+        position: 3
       });
       const payload = JSON.parse(sendBeacon.mock.calls[0][1]);
 
       expect(payload).toEqual({
         events: [
           expect.objectContaining({
             eventType: "click",
-            objectID: "1",
-            userID: "42",
+            eventName: "my-event",
+            indexName: "my-index",
+            objectID: ["1"],
+            position: [3],
+            userID: "mock-user-id",
             timestamp: expect.any(Number)
           })
         ]
       });
     });
   });
+
+  describe("eventName", () => {
+    it("should throw if no eventName passed", () => {
+      expect(() => {
+        (AlgoliaInsights as any).sendEvent("click", {
+          indexName: "my-index"
+        });
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"expected required parameter \`eventName\` to be a string"`
+      );
+    });
+    it("should throw if no eventName passed", () => {
+      expect(() => {
+        (AlgoliaInsights as any).sendEvent("click", {
+          eventName: 3,
+          indexName: "my-index"
+        });
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"expected required parameter \`eventName\` to be a string"`
+      );
+    });
+  });
+
+  describe("indexName", () => {
+    it("should throw if no indexName passed", () => {
+      expect(() => {
+        (AlgoliaInsights as any).sendEvent("click", {
+          eventName: "my-event"
+        });
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"expected required parameter \`indexName\` to be a string"`
+      );
+    });
+    it("should throw if no indexName is not a string", () => {
+      expect(() => {
+        (AlgoliaInsights as any).sendEvent("click", {
+          eventName: "my-event",
+          indexName: 2
+        });
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"expected required parameter \`indexName\` to be a string"`
+      );
+    });
+  });
+
+  describe("objectID and position", () => {
+    it("should support single objectID and position", () => {
+      (AlgoliaInsights as any).sendEvent("click", {
+        eventName: "my-event",
+        indexName: "my-index",
+        objectID: "1",
+        position: 3
+      });
+      expect(XMLHttpRequest.send).toHaveBeenCalledTimes(1);
+      const payload = JSON.parse(XMLHttpRequest.send.mock.calls[0][0]);
+      expect(payload).toEqual({
+        events: [
+          expect.objectContaining({
+            objectID: ["1"],
+            position: [3]
+          })
+        ]
+      });
+    });
+    it("should support multiple objectID and position", () => {
+      (AlgoliaInsights as any).sendEvent("click", {
+        eventName: "my-event",
+        indexName: "my-index",
+        objectID: ["1", "2"],
```

Comment:
What about support only the plural form?

Reply:
this is answered [here](https://github.com/algolia/search-insights.js/pull/22#discussion_r238710452)